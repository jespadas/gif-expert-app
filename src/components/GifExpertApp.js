import React, { useState } from 'react';

import { AddCategory } from './AddCategory';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    // Usando .concat
    /* const handleAdd = () => {
        setCategories(categories => categories.concat("New category " + Math.random(1)));
    } */

    // Usando Spread ...
    /* const handleAdd = () => {
        setCategories(categories => [...categories, "New category "]);
    } */

    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory />
            <hr />



            <ul>
                {categories.map((category) => {
                    return <li key={category}>{category}</li>
                })}
            </ul>

        </>
    )
}