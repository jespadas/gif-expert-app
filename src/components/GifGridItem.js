import React from 'react'

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="animate__animated animate__zoomIn">
            <img
                alt={title}
                src={url}
            />
            <p>{title}</p>
        </div>
    )
}
