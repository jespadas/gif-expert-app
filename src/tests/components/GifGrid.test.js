import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en GifGrid.js', () => {

    const category = "Back to the future";

    test('Debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan imagenes (useFetchGifs)', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://localhost:3000/img.png',
            title: 'The gif'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);

        // Comparando con snapshot
        expect(wrapper).toMatchSnapshot();
        // Buscando si existe un p en el componente
        expect(wrapper.find('p').exists()).toBe(true);
        // Comprobando el numero de elementos retornados
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })


});
