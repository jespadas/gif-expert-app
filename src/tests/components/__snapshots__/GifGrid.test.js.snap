// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid.js Debe mostrar items cuando se cargan imagenes (useFetchGifs) 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
    Back to the future
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    LOADING...
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="The gif"
      url="http://localhost:3000/img.png"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid.js Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
    Back to the future
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    LOADING...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
